app/assets/javascripts/discourse/components/site-header.js.es6

```diff
@@ -23,6 +30,146 @@ const SiteHeaderComponent = MountWidget.extend(Docking, {
     this.queueRerender();
   },
 
+  _animateOpening($panel) {
+    $panel.css({ right: "", left: "" });
+    this._panMenuOffset = 0;
+  },
+
+  _animateClosing($panel, menuOrigin, windowWidth) {
+    $panel.css(menuOrigin, -windowWidth);
+    this._animate = true;
+    Ember.run.schedule("afterRender", () => {
+      this.eventDispatched("dom:clean", "header");
+      this._panMenuOffset = 0;
+    });
+  },
+
+  _handlePanDone(offset, event) {
+    const velocity = 40;
+    const $window = $(window);
+    const windowWidth = parseInt($window.width());
+    const $menuPanels = $(".menu-panel");
+    const menuOrigin = this._panMenuOrigin;
+    this._shouldMenuClose(event, menuOrigin)
+      ? (offset += velocity)
+      : (offset -= velocity);
+    $menuPanels.each((idx, panel) => {
+      const $panel = $(panel);
+      const $headerCloak = $(".header-cloak");
+      $panel.css(menuOrigin, -offset);
+      $headerCloak.css("opacity", Math.min(0.5, (300 - offset) / 600));
+      if (offset > windowWidth) {
+        this._animateClosing($panel, menuOrigin, windowWidth);
+      } else if (offset <= 0) {
+        this._animateOpening($panel);
+      } else {
+        //continue to open or close menu
+        this._scheduledMovingAnimation = window.requestAnimationFrame(() =>
+          this._handlePanDone(offset, event)
+        );
+      }
+    });
+  },
+
+  _shouldMenuClose(e, menuOrigin) {
+    // menu should close after a pan either:
+    // if a user moved the panel closed past a threshold and away and is NOT swiping back open
+    // if a user swiped to close fast enough regardless of distance
+    const distanceThreshold = 200;
+    const velocityThreshold = 0.1;
+    if (menuOrigin === "right") {
+      return (
+        (e.deltaX > distanceThreshold && e.velocityX > -velocityThreshold) ||
+        e.velocityX > velocityThreshold
+      );
+    } else {
+      return (
+        (e.deltaX < -distanceThreshold && e.velocityX < velocityThreshold) ||
+        e.velocityX < -velocityThreshold
+      );
+    }
+  },
+
+  panStart(e) {
+    //android supports pulling in from the screen edges
+    const screenEdgeMargin = 30;
```

Comment:
I would recommend pulling these out and putting them at the top of the file like `SCREN_EDGE_MARGIN = 30`. It's nice to collect those kinds of things up top in case you want to export them later.